src/dos/drive_local.cpp

```diff
@@ -70,8 +72,36 @@ bool localDrive::IsFirstEncounter(const std::string& filename) {
 	return was_inserted;
 }
 
-bool localDrive::FileOpen(DOS_File** file, char * name, Bit32u flags) {
-	const char* type = nullptr;
+// Search the Files[] inventory for an open file matching the requested
+// local drive and file name. Returns nullptr if not found.
+DOS_File *FindOpenFile(const DOS_Drive *drive, const char *name)
+{
+	if (!drive || !name)
+		return nullptr;
+
+	uint8_t drive_num = DOS_DRIVES; // default out range
+	// Look for a matching drive mount
+	for (uint8_t i = 0; i < DOS_DRIVES; ++i) {
+		if (Drives[i] && Drives[i] == drive) {
+			drive_num = i;
+			break;
+		}
+	}
+	if (drive_num == DOS_DRIVES) // still out of range, no matching mount
+		return nullptr;
+
+	// Look for a matching open filename on the same mount
+	for (uint8_t i = 0; i < DOS_FILES; ++i)
```

Comment:
Usage of `uint8_t` in here raised an eyebrow for me; turns out DOSBox file limit of 127 per directory is incorrect…

https://www.datadoctor.biz/data_recovery_programming_book_chapter3-page3.html

This was a limit of FAT12 used on floppy disks; FAT16 supports 512 files per directory.

How about:
```
for (auto *file : Files) {
        if (file && file->IsOpen() …
}
```
?

This way we don't need to worry about particular type used for iteration here.

Reply:
Amazing catch @dreamer.  Little flaws like this could be the smoking gun for why some games have only partially worked when using a "local" mount, but become fully working when using a HDD FAT16 image (when the game itself doesn't know the difference.. in both cases it's hitting dosbox's filesystem API.. just two difference buckets of code).

Reply:
The `Files[]` array holds the set of open handles and is global across all the mounts; so 127 is here equivalent to setting `FILES=127` in `config.sys`.